### PR TITLE
[CWS] Add unit tests for activity_tree.CreateProcessNode func

### PR DESCRIPTION
**Pull request by spikat**
### What does this PR do?

This PR adds 680 unit tests for activity_tree.CreateProcessNode (thanks to some automatic matrix).
NB: sadly, the tests were added on a separate directory to avoid cycling imports.

Also, add some more tests for profile.tryAutolearn.

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature. 
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: pkg/security/security_profile/tests/activity_tree_test.go [hunk 1]
```diff
@@ -0,0 +1,635 @@
...
+			resetActivityTree:     true,
+			parentProcessPath:     "/bin/runc",
+			processPath:           "/bin/bar",
+			completeLineage:       true,
+			differentContainerID:  false,
+			fileLessParent:        true,
+			resultNodeShouldBeNil: false,
+			resultNewProcessNode:  true,
+			resultErr:             nil,
+		},
```
Comment (spikat) [inline pkg/security/security_profile/tests/activity_tree_test.go:544]: I'm still not sure about this behavior: having a fileless runc added as root node without errors

Comment (spikat) [reply pkg/security/security_profile/tests/activity_tree_test.go]: Logic updated after discussion

File: pkg/security/security_profile/activity_tree/activity_tree.go [hunk 2]
```diff
@@ -372,11 +381,9 @@ func (at *ActivityTree) CreateProcessNode(entry *model.ProcessCacheEntry, genera
 	// find or create a ProcessActivityNode for the parent of the input ProcessCacheEntry. If the parent is a fork entry,
 	// jump immediately to the next ancestor.
 	parentNode, newProcessNode, err := at.CreateProcessNode(GetNextAncestorBinaryOrArgv0(&entry.ProcessContext), Snapshot, dryRun)
-	if err != nil || (newProcessNode && dryRun) {
+	if newProcessNode && dryRun {
```
Comment (safchain) [inline pkg/security/security_profile/activity_tree/activity_tree.go:384]: why we don't check the error anymore ?

File: pkg/security/security_profile/dump/activity_dump.go [hunk 3]
```diff
@@ -308,9 +308,11 @@ func (ad *ActivityDump) SetTimeout(timeout time.Duration) {
 func (ad *ActivityDump) updateTracedPid(pid uint32) {
 	// start by looking up any existing entry
 	var cookie uint32
-	_ = ad.adm.tracedPIDsMap.Lookup(pid, &cookie)
-	if cookie != ad.LoadConfigCookie {
-		_ = ad.adm.tracedPIDsMap.Put(pid, ad.LoadConfigCookie)
+	if ad.adm != nil { // it could be nil when running unit tests
```
Comment (safchain) [inline pkg/security/security_profile/dump/activity_dump.go:311]: could we create an adm in the UT

**Review by safchain — APPROVED** [review]
two comments otherwise LGTM

File: pkg/security/security_profile/dump/activity_dump.go (hunk 3 above)
Comment (spikat) [reply pkg/security/security_profile/dump/activity_dump.go:311]: Not easily, and to make it works it should also have the tracePIDsMap initialized and IMHO I think that defeat the purpose of having simple UT with limited scope

**Pull request merged**